src/pluggy/_manager.py

```diff
@@ -181,7 +192,20 @@ def parse_hookimpl_opts(self, plugin: _Plugin, name: str) -> HookimplOpts | None
         customize how hook implementation are picked up. By default, returns the
         options for items decorated with :class:`HookimplMarker`.
         """
-        method: object = getattr(plugin, name)
+
+        if _attr_is_property(plugin, name):
+            # @property methods can have side effects, and are never hookimpls
+            return None
+
+        method: object
+        try:
+            method = getattr(plugin, name)
+        except AttributeError:
```

Comment:
If you're able to cook a test for this case that would be nice.

Reply:
see comment here: https://github.com/pytest-dev/pluggy/pull/536#discussion_r1942162516